pkg/util/validation/validation.go

```diff
@@ -29,6 +32,18 @@ func ValidatePropagationSpec(spec policyv1alpha1.PropagationSpec) field.ErrorLis
 	return allErrs
 }
 
+// ValidateResourceSelectors validates all resourceSelectors before creation or update.
+func ValidateResourceSelectors(resourceSelectors []policyv1alpha1.ResourceSelector, fldPath *field.Path) field.ErrorList {
```

Comment:
The function name doesn't match its responsibility. This function is actually focused on a limited scenario and is not general-purpose.